src/common/transformations/src/transformations/common_optimizations/activations_scaling.cpp

```diff
@@ -0,0 +1,222 @@
+                                                                    inverse_scale_const_f16 : inverse_scale_const_f32;
+                    auto scale_down = std::make_shared<ov::op::v1::Multiply>(dep.get_source_output(),
+                                                                             inverse_scale_const->output(0));
+                    dep.replace_source_output(scale_down->output(0));
+                }
+            }
+        }
+
+        auto multiply = std::dynamic_pointer_cast<ov::op::v1::Multiply>(node);
+        auto matmul = std::dynamic_pointer_cast<ov::op::v0::MatMul>(node);
+        if ((multiply || matmul) && num_scaled_down_inputs == 2) {
+            auto dep = node->input(1);
+            std::shared_ptr<ov::Node> scale_const = (dep.get_element_type() == ov::element::f16) ?
+                                                    scale_const_f16 : scale_const_f32;
+            auto scale_up = std::make_shared<ov::op::v1::Multiply>(dep.get_source_output(), scale_const->output(0));
+            dep.replace_source_output(scale_up->output(0));
+        }
+
+        auto sdpa = std::dynamic_pointer_cast<ov::op::v13::ScaledDotProductAttention>(node);
+        if (sdpa) {
+            for (size_t i = 0; i < 2; i++) {
+                if (scaled_down_nodes.find(node->input(i).get_source_output().get_node_shared_ptr()->get_friendly_name()) != scaled_down_nodes.end()) {
+                    std::shared_ptr<ov::Node> scale_const = (node->get_input_element_type(i) == ov::element::f16) ? scale_const_f16 : scale_const_f32;
+                    auto transpose = std::dynamic_pointer_cast<ov::op::v1::Transpose>(node->input(i).get_source_output().get_node_shared_ptr());
+                    if (transpose) {
+                        auto scale_up = std::make_shared<ov::op::v1::Multiply>(transpose->get_input_source_output(0),
+                                                                               scale_const->output(0));
+                        transpose->input(0).replace_source_output(scale_up->output(0));
+                    } else {
+                        auto scale_up = std::make_shared<ov::op::v1::Multiply>(node->get_input_source_output(i),
+                                                                               scale_const->output(0));
+                        node->input(i).replace_source_output(scale_up->output(0));
+                    }
+                }
+            }
+
+            if (scaled_down_nodes.find(node->input(2).get_source_output().get_node_shared_ptr()->get_friendly_name()) != scaled_down_nodes.end()) {
+                scaled_down_nodes.insert(node->get_friendly_name());
+            }
+            continue;
+        }
+
+        // input(scaled_down) -- activation
+        // ==>
+        // input(scaled_down) -- convert(precision_up) -- multiply(scale_up) -- activation -- multiply(scale_down) -- convert(precision_down)
```

Comment:
I didn't get why you do up convert for activation (and only those types). Could you explain the motivation? if we really need to compute those in f32 precision, then I think up-convert should be a part of ConvertPrecision(f16) transformation (which handles mixed precision)

Reply:
This part is also removed.